pensieve/analysis.py

```diff
@@ -160,10 +144,59 @@ def run(self, current_date: datetime, dry_run: bool):
             return
 
         self.logger.info("Executing query for %s", self.experiment.slug)
-        self.bigquery.execute(sql, res_table_name)
+        result = self.bigquery.execute(sql, res_table_name)
         self._publish_view("week")
         self.logger.info("Finished running query for %s", self.experiment.slug)
 
+        return result.to_dataframe()
```

Comment:
Can we make the statistics-running code responsible for fetching the results?

"Compute the statistics for this table" sounds like a better abstraction than "compute the statistics for this in-memory dataframe"; it'll be easier to hand it off to another process.

Reply:
That would certainly be possible.